src/hooks/useForm.jsx

```diff
@@ -0,0 +1,93 @@
+import { useCallback, useRef } from 'react';
+
+/**
+ * @name useForm
+ * @description A hook providing several method to control forms.
+ *
+ * @author Timothée Simon-Franza
+ */
+const useForm = () => {
+	const inputsRefs = useRef({});
+
+	/**
+	 * @function
+	 * @name getFormValues
+	 * @description Returns the value of all controlled fields as an object.
+	 *
+	 * @author Timothée Simon-Franza
+	 *
+	 * @returns {object}
+	 */
+	const getFormValues = () => {
+		const formValues = Object.values(inputsRefs.current)
+			.map(({ element: { value }, name }) => ([name, value]));
+
+		return Object.fromEntries(formValues);
+	};
+
+	/**
+	 * @function
+	 * @name registerFormField
+	 * @description A callback method used by controlled form fields to register themselves to the form.
+	 *
+	 * @author Timothée Simon-Franza
+	 *
+	 * @param {object} formFieldRef: The ref to register.
+	 */
+	const registerFormField = useCallback((formFieldRef) => {
+		if (formFieldRef.name) {
+			inputsRefs.current[formFieldRef.name] = formFieldRef;
```

Comment:
maybe an else clause displaying a warning in the console could be useful, like `Warning: attempt to register a form field without a name property`